packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,342 @@
+          setCurrentSelection('mumbai');
+        }}
+      />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+### Approach Comparisons
+
+As mentioned in [Propsed Approaches](#proposed-approaches), we considered 3 approaches majorly -
+
+- `onChange` Prop-Based
+  - [`onChange` + `value`](#onchange--value)
+  - [`onChange` + `isSelected`](#onchange--isselected-)
+- Hook-Based ([`useSelect`](#useselect))
+
+#### Hook-Based vs `onChange` Prop-Based
+
+First we'll discuss Prop-Based vs Hook-Based because throughout initial discussion we talked about creating hook-based API but following points made me change my opinion-
+
+|                             | Hooks-Based Approach                                                                    | ✅ `onChange` Prop Based Approach                                                     |
+| --------------------------- | --------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------- |
+| **Occurences in Ecosystem** | Rare (only seen in downshift-js)                                                        | ✅ Common (Almost all libraries use some form of this approach for controlled select) |
+| **Learning Curve**          | More (Introduces new hook so need documentation on input type, output type, usage, etc) | ✅ Less (Only adds one prop on top of existing implementation)                        |
+| **API Verbosity**           | ✅ Less verbose                                                                         | Comparitively more verbose but not too complex ([Example](#onchange--isselected-))    |
+
+**Conclusion**: We are going with `onChange` prop-based approach ✅
+
+#### `onChange` + `value` vs `onChange` + `isSelected`
+
+Lets compare 2 `onChange` prop-based approaches now-
+
+| `onChange` + `isSelected`                                                                    | `onChange` + `value`                                                       |
+| -------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------- |
+| Preferred Approach if we say `ActionListItem` is selectable component independent of trigger | Preferred Approach if only `SelectInput` trigger has selectable components |
+
+**Conclusion:** We decided that we would need a combination of both approaches. Because-
+
+- Based on the [selectable usecase of Menu](#menu-with-selected-item), the ActionListItem has to be "selectable" individually. Which means we definitely need something like `isSelected` on ActionListItem.
+- However the concerns raised with `onChange` + `isSelected` approach were that the controlled states end up on 2 different indepdent component which is confusing and they in a way become bound to each other.
```

Comment:
elaborate this line for the simplicity of readers as to what we mean by independent component. maybe take an example that `onChange` on select input and isSelected on `ActionListItem`

Reply:
there's another downside as well that we discussed and we shall add here, where if we have isSelected then consumers will have to always mention that when consuming this prop so poor DX.

Reply:
done